scripts/create_synthetic_population.R

```diff
@@ -0,0 +1,106 @@
+## =================================#
+## Setup ---------------
+## =================================#
+library(tidyverse)
+library(tigris)
+library(tidycensus)
+library(patchwork)
+
+set.seed(1234)
+
+state_synth <- "WY"
+year_synth <- 2023
+population_size <- 1000
+
+## =================================#
+## Get population ---------------
+## =================================#
+pums_vars <- pums_variables |>
+  filter(year == 2018, survey == "acs1") |>
+  distinct(var_code, var_label, data_type, level)
+
+person_variables <- c(
+  "SPORDER", "SERIALNO", "PWGTP",
+  "AGEP", "SEX", "PUMA", "REGION"
+)
+house_variables <- c("WGTP", "NP")
+
+sample_pums <- get_pums(
+  variables = c(person_variables, house_variables),
+  state = state_synth,
+  survey = "acs1",
+  year = year_synth
+)
+
+household_pums <- sample_pums |>
+  dplyr::select(SERIALNO, all_of(house_variables)) |>
+  distinct()
+
+## =================================#
+## Create population ---------------
+## =================================#
+synth_pop_df <- tibble()
+house_counter <- 0
+while (nrow(synth_pop_df) < population_size) {
+  house_counter <- house_counter + 1
+  synth_pop_df <- bind_rows(synth_pop_df, household_pums) |>
+    sample_n(1, weight = WGTP) |>
+    left_join(sample_pums, by = (c("SERIALNO", "WGTP", "NP"))) |>
+    mutate(house_number = house_counter)
+}
```

Comment:
Does it matter that this can produce a data frame that doesn't necessarily have exactly the `population_size` but only at least that size? I don't think so since we can calculate the population later when using the synthetic data, but wanted to draw attention to the potential variability. I could really only see this mattering for population sizes that are small enough to be within the distribution of residents per household

Reply:
If we get fancier, maybe we can fix that. But for how simple this is, I don't think we should care too much about that.